cdap-standalone/src/main/java/co/cask/cdap/TrackerAppCreationService.java

```diff
@@ -0,0 +1,179 @@
+                                   ApplicationLifecycleService applicationLifecycleService,
+                                   ProgramLifecycleService programLifecycleService) {
+    this.cConf = cConf;
+    this.artifactRepository = artifactRepository;
+    this.applicationLifecycleService = applicationLifecycleService;
+    this.programLifecycleService = programLifecycleService;
+    this.applicationId = new ApplicationId(NamespaceId.DEFAULT.getNamespace(), "_Tracker");
+    this.auditFlow = new ProgramId(applicationId.getNamespace(), applicationId.getApplication(), ProgramType.FLOW,
+                                   "AuditLogFlow");
+    this.auditService = new ProgramId(applicationId.getNamespace(), applicationId.getApplication(), ProgramType.SERVICE,
+                                      "AuditLog");
+  }
+
+  @Override
+  protected void run() throws Exception {
+    try {
+      // Wait until tracker artifact is available
+      Tasks.waitFor(true, new Callable<Boolean>() {
+        @Override
+        public Boolean call() throws Exception {
+          List<ArtifactSummary> artifacts = null;
+          try {
+            artifacts = artifactRepository.getArtifacts(NamespaceId.SYSTEM, "tracker");
+          } catch (ArtifactNotFoundException ex) {
+            // expected, so suppress it
+          }
+          return artifacts != null && artifacts.size() > 0;
+        }
+      }, 5, TimeUnit.MINUTES, 5, TimeUnit.SECONDS, "Waiting for Tracker Artifact to be listed.");
+      // Find the latest tracker artifact based on the version
+      List<ArtifactSummary> artifacts = new ArrayList<>(artifactRepository.getArtifacts(NamespaceId.SYSTEM, "tracker"));
+      artifactSummary = artifacts.remove(0);
+      for (ArtifactSummary artifact : artifacts) {
+        ArtifactVersion old = new ArtifactVersion(artifactSummary.getVersion());
+        ArtifactVersion current = new ArtifactVersion(artifact.getVersion());
+        if (current.compareTo(old) > 0) {
+          artifactSummary = artifact;
+        }
+      }
+      createAndStartTracker();
+    } catch (Exception ex) {
+      // Not able to create Tracker App is not catastrophic, hence don't propagate exception
+      LOG.warn("Got an exception while trying to create and start Tracker App. ", ex);
+    }
+  }
+
+  private void createAndStartTracker() throws Exception {
+    TrackerAppConfig appConfig = getTrackerAppConfig();
+    applicationLifecycleService.deployApp(
+      Id.Namespace.from(applicationId.getNamespace()), applicationId.getApplication(),
+      Id.Artifact.from(Id.Namespace.SYSTEM, artifactSummary.getName(), artifactSummary.getVersion()),
+      GSON.toJson(appConfig), new TrackerProgramTerminator(programLifecycleService));
+    try {
+      programLifecycleService.start(auditFlow, ImmutableMap.<String, String>of(), false);
+    } catch (IOException ex) {
+      // Might happen if the program is being started in parallel through UI
```

Comment:
Can you log a debug with the ex.getMessage() there. You might not need to log the stacktrace.